crates/provider/src/builder.rs

```diff
@@ -144,8 +144,11 @@ impl<L, N> ProviderBuilder<L, Identity, N> {
     /// Add nonce management to the stack being built.
     ///
     /// See [`NonceFiller`]
-    pub fn with_nonce_management(self) -> ProviderBuilder<L, JoinFill<Identity, NonceFiller>, N> {
-        self.filler(NonceFiller::default())
+    pub fn with_nonce_management<M: NonceManager>(
+        self,
+        nonce_manager: M,
+    ) -> ProviderBuilder<L, JoinFill<Identity, NonceFiller<M>>, N> {
```

Comment:
I'm fine with this but I'd like a new function that calls this function the previous nonce manager (now called CachedNonce I believe)

Reply:
ok, so you want something like `with_{simple, cached}_nonce(_management) `methods?

Reply:
yeah, I think def for the builtin cached one